sklearn/preprocessing/_encoders.py

```diff
@@ -439,6 +608,185 @@ def _compute_drop_idx(self):
+            # stable sort to preserve original count order
+            smallest_levels = np.argsort(category_count, kind="mergesort")[
+                : -self.max_categories + 1
+            ]
+            infrequent_mask[smallest_levels] = True
+
+        output = np.flatnonzero(infrequent_mask)
+        if output.size == category_count.size:
+            raise ValueError(
+                f"All categories in column {col_idx} are "
+                "infrequent, try decreasing min_frequency"
+            )
+        return output if output.size > 0 else None
+
+    def _fit_infrequent_category_mapping(self, n_samples, category_counts):
+        """Fit infrequent categories.
+
+        Defines the private attribute: `_default_to_infrequent_mappings`. For
+        feature `i`, `_default_to_infrequent_mappings[i]` defines the mapping
+        from the integer encoding returned by `super().transform()` into
+        infrequent categories. If `_default_to_infrequent_mappings[i]` is None,
+        there were no infrequent categories in the training set.
+
+        For example if categories 0, 2 and 4 were frequent, while categories
+        1, 3, 5 were infrequent for feature 7, then these categories are mapped
+        to a single output:
+        `_default_to_infrequent_mappings[7] = array([0, 3, 1, 3, 2, 3])`
+
+        Parameters
+        ----------
+        n_samples : int
+            Number of samples in training set.
+        category_counts: list of ndarray
+            List of counts corresponding where `category_counts[i]` are the
+            counts for each category in `self.categories_[i]`.
+        """
+        self.infrequent_indices_ = [
+            self._identify_infrequent(category_count, n_samples, col_idx)
+            for col_idx, category_count in enumerate(category_counts)
+        ]
+
+        # compute mapping from default mapping to infrequent mapping
+        self._default_to_infrequent_mappings = []
+
+        for cats, infreq_idx in zip(self.categories_, self.infrequent_indices_):
+            # no infrequent categories
+            if infreq_idx is None:
+                self._default_to_infrequent_mappings.append(None)
+                continue
+
+            n_cats = len(cats)
+            # infrequent indices exist
+            mapping = np.empty(n_cats, dtype=np.int64)
+            n_infrequent_cats = infreq_idx.size
+
+            # infrequent categories are mapped to the last element.
+            n_frequent_cats = n_cats - n_infrequent_cats
+            mapping[infreq_idx] = n_frequent_cats
+
+            frequent_indices = np.setdiff1d(np.arange(n_cats), infreq_idx)
+            mapping[frequent_indices] = np.arange(n_frequent_cats)
+
+            self._default_to_infrequent_mappings.append(mapping)
+
+    def _map_to_infrequent_categories(self, X_int, X_mask):
+        """Map categories to infrequent categories. This modifies X_int
```

Comment:
Really we're mapping the infrequent categories to the "infrequent" category, right?